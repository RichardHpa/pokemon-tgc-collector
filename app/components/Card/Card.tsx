import type { FC, ReactNode } from "react";

interface CardProps {
  children: ReactNode;
}

export const Card: FC<CardProps> = ({ children }) => {
  return (
    <div className="h-full rounded-lg border border-gray-200 bg-white p-4 shadow-md dark:border-gray-700 dark:bg-gray-800">
      {children}
    </div>
  );
};
