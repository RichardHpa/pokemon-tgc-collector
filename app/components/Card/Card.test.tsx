import { render, screen } from "@testing-library/react";

import { Card } from "./Card";

describe("Card", () => {
  test("renders Card", () => {
    render(<Card>Card</Card>);
    expect(screen.getByText("Card")).toBeInTheDocument();
  });

  test("renders Card snapshot", () => {
    const { container } = render(<Card>Test</Card>);
    expect(container).toMatchSnapshot();
  });
});
