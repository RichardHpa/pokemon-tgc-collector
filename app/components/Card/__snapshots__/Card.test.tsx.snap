// Vitest Snapshot v1

exports[`Card > renders Card snapshot 1`] = `
<div>
  <div
    class="h-full rounded-lg border border-gray-200 bg-white p-4 shadow-md dark:border-gray-700 dark:bg-gray-800"
  >
    Test
  </div>
</div>
`;
